paasta_tools/metrics/metrics_lib.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import time
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from typing_extensions import Protocol
+
+from paasta_tools.utils import load_system_paasta_config
+
+log = logging.getLogger(__name__)
+
+try:
+    import yelp_meteorite
+except ImportError:
+    yelp_meteorite = None
+
+_metrics_interfaces: Dict[str, Type['BaseMetrics']] = {}
+
+
+class TimerProtocol(Protocol):
+    def start(self) -> None:
+        ...
+
+    def stop(self) -> None:
+        ...
+
+
+class GaugeProtocol(Protocol):
+    def set(self, value: Union[int, float]) -> None:
+        ...
+
+
+class BaseMetrics(Protocol):
```

Comment:
can you make this a real abstract base class? https://docs.python.org/3/library/abc.html